plotbitrate.py

```diff
@@ -28,283 +29,596 @@
-
-# terminate progress line
-if args.progress:
-    print(flush=True)
-
-# setup new figure
-matplot.figure().canvas.set_window_title(args.input)
-matplot.title("Stream Bitrate vs Time")
-matplot.xlabel("Time (sec)")
-matplot.ylabel("Frame Bitrate (kbit/s)")
-matplot.grid(True)
-
-# map frame type to color
-frame_type_color = {
-    # audio
-    'A': 'red',
-    # video
-    'I': 'red',
-    'P': 'green',
-    'B': 'blue'
-}
-
-global_peak_bitrate = 0.0
-global_mean_bitrate = 0.0
-
-# render charts in order of expected decreasing size
-for frame_type in ['I', 'P', 'B', 'A']:
-
-    # skip frame type if missing
-    if frame_type not in bitrate_data:
-        continue
-
-    # convert list of tuples to numpy 2d array
-    frame_list = bitrate_data[frame_type]
-    frame_array = numpy.array(frame_list)
-
-    # update global peak bitrate
-    peak_bitrate = frame_array.max(0)[1]
-    if peak_bitrate > global_peak_bitrate:
-        global_peak_bitrate = peak_bitrate
-
-    # update global mean bitrate (using piecewise mean)
-    mean_bitrate = frame_array.mean(0)[1]
-    global_mean_bitrate += mean_bitrate * (len(frame_list) / frame_count)
-
-    # plot chart using gnuplot-like impulses
-    matplot.vlines(
-        frame_array[:,0], [0], frame_array[:,1],
-        color=frame_type_color[frame_type],
-        label="{} Frames".format(frame_type))
-
-# set y-axis limits if requested
-if args.min:
-    matplot.ylim(ymin=args.min)
-if args.max:
-    matplot.ylim(ymax=args.max)
-
-# calculate peak line position (left 15%, above line)
-peak_text_x = matplot.xlim()[1] * 0.15
-peak_text_y = global_peak_bitrate + \
-    ((matplot.ylim()[1] - matplot.ylim()[0]) * 0.015)
-peak_text = "peak ({:.0f})".format(global_peak_bitrate)
-
-# draw peak as think black line w/ text
-matplot.axhline(global_peak_bitrate, linewidth=2, color='black')
-matplot.text(peak_text_x, peak_text_y, peak_text,
-    horizontalalignment='center', fontweight='bold', color='black')
-
-# calculate mean line position (right 85%, above line)
-mean_text_x = matplot.xlim()[1] * 0.85
-mean_text_y = global_mean_bitrate + \
-    ((matplot.ylim()[1] - matplot.ylim()[0]) * 0.015)
-mean_text = "mean ({:.0f})".format(global_mean_bitrate)
-
-# draw mean as think black line w/ text
-matplot.axhline(global_mean_bitrate, linewidth=2, color='black')
-matplot.text(mean_text_x, mean_text_y, mean_text,
-    horizontalalignment='center', fontweight='bold', color='black')
-
-matplot.legend()
-
-# render graph to file (if requested) or screen
-if args.output:
-    matplot.savefig(args.output, format=args.format)
-else:
-    matplot.show()
-
-# vim: ai et ts=4 sts=4 sw=4
+            values_min = sums_of_values
+            values_max = [
+                sum(pair) for pair in zip(sums_of_values, values)
+            ]
+        sums_of_values = values_max
+        color = Color[frame_type].value if frame_type in dir(Color) \
+            else Color.FRAME.value
+        bars[frame_type] = matplot.fill_between(
```

Comment:
If you add `zorder=2` whenever calling `matplot.fill_between()` (also below) then it will put the y-axis grid lines below the graph which makes it look nicer.